### DISCUSSION TITLE
Unauthenticated paths?

### DISCUSSION THREAD

**Discussion by aureateflux** [mealie-recipes/mealie#2412] [Q&A]
I'm playing around with using Authentik to add an authentication layer on top of Mealie so I can enforce some MFA. I didn't find any substantial discussion anywhere about doing this with Mealie, so I'm just following the typical flow of adding applications to Authentik.

I have the authentication working (it still requires a login to mealie after logging into Authentik, of course, since forward auth isn't implemented yet), and I haven't found any problems running it while signed into Authentik.

Sharing doesn't work, however, because the identity provider needs to know which paths don't need to be authenticated. I tried adding ^/shared/.* to the provider's unauthenticated paths list. That at least gets the page starting to load, but it doesn't load the actual recipe.

I'm hoping someone who has experience using Authentik or Authelia with Mealie can help me with identifying all the unauthenticated paths that need to be bypassed. There must be a path i'm missing that's causing the conflict. When I inspect the page, I'm seeing that it's refusing to run the <URL> script because of CSP rules I have in place. If I add the authentik domain path as a trusted source in the CSP headers, it triggers cross origin read blocking.

Any ideas on what path I need to tell Authentik to bypass so the script stays in the subdomain in this case?

**Comment by lenderom** [comment]
I can confirm, this behavior still exists. I tried to implement the same thing with Authelia and failed.

The problem seems to be, that the "shared/public copy" fetches alot of data from different paths. When looking into the browsers  network tab, this becomes clear.

That's how it looks with middelware authentication layer without access:
<img width="707" height="146" alt="image" src="https://github.com/user-attachments/assets/b1593aaf-afff-4ccc-8fb0-bdd850abda8d" />

And thats how it looks with middelware authentication layer with access:
<img width="595" height="295" alt="image" src="https://github.com/user-attachments/assets/1c32e8a4-2c90-4b60-a914-8de3daafdc15" />
(There are even more files fetched)

So I guess I would technically work to bypass all of these paths like:

`/_nuxt/`
`/fonts/`
`/api/`
`/icons/`
and so on. But that not helpful with the intention to improve security.

@michael-genson  Do you think it would be possible to rework the Share-Feature so most of the needed files are directly fetched from the share path? So like a more strict separation between apis for authenticated users and public available links fo receipes?
